php-nginx/detect_php_version.php

```diff
@@ -18,32 +18,49 @@
 
 use Composer\Semver\Semver;
 
-// Available versions in the order we want to check.
-$available_versions = [
-    getenv('PHP71_VERSION'),
-    getenv('PHP70_VERSION'),
-    getenv('PHP56_VERSION'),
-];
+class DetectPhpVersion
+{
+    public static function versionFromComposer($filename, $availableVersions = null)
+    {
+        $composer = json_decode(file_get_contents($filename), true);
+        if (is_array($composer)
+            && array_key_exists('require', $composer)
+            && array_key_exists('php', $composer['require'])) {
+            $constraints = $composer['require']['php'];
+            return self::version($constraints, $availableVersions);
+        }
+        return '';
+    }
+
+    public static function version($constraint, $availableVersions = null)
+    {
+        $availableVersions = $availableVersions ?: self::detectAvailableVersions();
+        foreach ($availableVersions as $version) {
+            if (Semver::satisfies($version, $constraint)) {
+                // The first match wins, picking the highest version possible.
+                return $version;
+            }
+        }
+        return '';
+    }
 
-if (count($argv) < 2) {
-    die("Usage:\n" . $argv[0] . " filename\n");
+    private static function detectAvailableVersions()
+    {
+        return [
+            getenv('PHP71_VERSION'),
```

Comment:
They already are out of date, this is just ported to a class to make testing it easier.